### PR TITLE
Added exchange_opts in config

### PR DESCRIPTION
**Pull request by indyone**
Added exchange_opts in queue config for optionally declaring an exchange before binding the queue on it.

### CODE REVIEW

File: lib/ex_rabbit_m_q/consumer.ex [hunk 1]
```diff
@@ -426,6 +428,22 @@ defmodule ExRabbitMQ.Consumer do
         end
       end
 
+      defp xrmq_exchange_declare(channel, %{exchange_opts: nil}) do
+        :ok # Ignore when not set
+      end
+
+      defp xrmq_exchange_declare(channel, %{exchange_opts: opts}) do
+        AMQP.Exchange.declare(channel, opts[:exchange], opts[:type] || :direct, opts)
```
Comment (bottlenecked) [inline lib/ex_rabbit_m_q/consumer.ex]: maybe call this opts.name instead?

File: lib/ex_rabbit_m_q/consumer.ex [hunk 2]
```diff
@@ -426,6 +428,22 @@ defmodule ExRabbitMQ.Consumer do
...
+      end
+
+      defp xrmq_queue_bind(channel, queue, %{bind_opts: nil}) do
+        :ok # Ignore when not set
+      end
+
+      defp xrmq_queue_bind(channel, queue, %{bind_opts: opts}) do
+        AMQP.Queue.bind(channel, queue, opts[:exchange], opts[:extra_opts] || [])
```
Comment (bottlenecked) [inline lib/ex_rabbit_m_q/consumer.ex]: we should break away from the bind_opts.exchange config and use the newer exchange_opts section.
Make sure to update to new minor version

File: lib/ex_rabbit_m_q/consumer.ex [hunk 3]
```diff
@@ -475,7 +493,8 @@ defmodule ExRabbitMQ.Consumer do
           queue: config[:queue],
           queue_opts: config[:queue_opts],
           consume_opts: config[:consume_opts],
-          bind_opts: config[:bind_opts],
+          exchange_opts: config[:exchange_opts],
+          bind_opts: config[:bind_opts]
```
Comment (bottlenecked) [inline lib/ex_rabbit_m_q/consumer.ex]: should we pull extra_opts out of bind_opts?

Comment (indyone) [reply lib/ex_rabbit_m_q/consumer.ex]: Ok, done. `bind_opts` now contain only the `options` for the `AMQP.Queue.Bind` function.

**Pull request merged**